src/Microsoft.VisualStudio.Threading/AsyncLazy`1.cs

```diff
@@ -185,7 +185,7 @@ public Task<T> GetValueAsync(CancellationToken cancellationToken)
 
             if (!this.value.IsCompleted)
             {
-                this.joinableTask?.JoinAsync(cancellationToken).Forget();
+                return this.joinableTask?.JoinAsync(cancellationToken) ?? this.value.WithCancellation(cancellationToken);
```

Comment:
This could be easier to grok if we add `&& this.joinableTask is not null` to the `if` condition and then change this line to simply:

```cs
                return this.joinableTask.JoinAsync(cancellationToken);
```

Reply:
In fact, I see `JoinableTask<T>.JoinAsync` already has a fast-path for completion, so maybe we can even drop the is-completion check and simply always use JoinAsync if `joinableTask is not null`.

Reply:
yes, we can drop the condition check, and that would simplify the code.

i didn't move this.joinableTask null check to the condition, because it will leave a race condition, unless we capture the this.joinableTask to a local variable. The '?.' did that automatically.  Originally, I thought checking the inner task IsComplete and the JTF task IsComplete is slightly different, where we have a task continuation on the inner task to set the JTF task to completion. (Although the difference is quite small, so it doesn't really make a difference.) 

I read the implementation again, and it appears that we set this.joinableTask to null right at the end of the JTF task in the AsyncLazy (it was to cut memory usage), so this.joinableTask must be null when this.value.IsCompleted to be true. With that, we don't really need repeat extra condition check here anyway.